Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -0,0 +1,212 @@
+        {
+            $ComponentKeyAndPathMap.Add($component.key, $component.path)
+        }
+    }
+}
+
+#
+# Creates a mapping of msbuild project guid and the path of .xxproj file on disk using ProjectInfo.xml file
+#
+function CreateProjectGuidToProjectPathMap
+{        
+    $parentFolder = GetSonarQubeOutDirectory
+    $parentFolderItem = Get-Item $parentFolder
+    $directories = $parentFolderItem.GetDirectories()
+
+    foreach ($directory in $directories)
+    {
+        $projectInfoFilePath = [System.IO.Path]::Combine($directory.FullName, "ProjectInfo.xml")
+        
+        if ([System.IO.File]::Exists($projectInfoFilePath))
+        {
+            Write-Verbose "CreateProjectGuidToProjectPathMap: Processing project info file: $projectInfoFilePath"
+            [xml]$xmlContent = Get-Content $projectInfoFilePath
+
+            if ($xmlContent -and !$ProjectGuidAndFilePathMap.ContainsKey($xmlContent.ProjectInfo.ProjectGuid))
+            {
+                $ProjectGuidAndFilePathMap.Add($xmlContent.ProjectInfo.ProjectGuid, $xmlContent.ProjectInfo.FullPath)
+            }
+        }
+    }
+}
+
+#
+# Expected format for component '[SonarQube project key]:[SonarQube project value]:[MSBuild project guid]:[file name relative to MSBuild project file path]'
+#
+function GetComponentGuid
+{
+    param ([ValidateNotNullOrEmpty()][string]$component)    
+  
+    $tokens = $component.ToString().Split(":")
+
+    if ($tokens.Count -ne 4) 
+    {
+        throw "Internal error: component $component is not in the expected format (expected 4 parts)"
+    }
+
+    #third token must be a guid
+    $guidToken = $tokens[2]
+
+    $outGuid = New-Object -TypeName "System.Guid"
+    if (![System.Guid]::TryParse($guidToken, [ref]$outGuid))
+    {
+        throw "Internal error: $guidToken is not a GUID"
+    }
+
+    return $guidToken
+}
+
+#
+# Returns a path relative to the repo root for a file which has new code analysis issue(s)
+#
+function GetPathRelativeToRepoRoot($component)
+{
+    if ($component -and $ComponentKeyAndRelativePathCache.ContainsKey($component))
+    {
+        $relativeFilePath = $ComponentKeyAndRelativePathCache[$component]
+        Write-Verbose "GetPathRelativeToRepoRoot: Found cached entry, returning data from cache, relativePath:$relativeFilePath"
+
+        return $relativeFilePath
+    }
+
+    #sonar runner creates the component value as '[SonarQube project key]:[SonarQube project value]:[MSBuild project guid]:[file name relative to MSBuild project file path]'
+   
+    #third token must be guid
+    $guidToken = $tokens[2]
+    Write-Verbose "GetPathRelativeToRepoRoot: guidToken:$guidToken"
+
+    if (!$ProjectGuidAndFilePathMap.ContainsKey($guidToken))
+    {
+        Write-Verbose "GetPathRelativeToRepoRoot: An entry for project guid $guidToken could not be found, check ProjectInfo.xml file"
+        return $null
+    }
+    if (!$ComponentKeyAndPathMap.ContainsKey($component))
```

Comment:
Why don't you need to check if £$component ? (like you did in 141). If it got here it could be null...